lib/pgBackRest/Protocol/Common/Master.pm

```diff
@@ -102,9 +93,14 @@ sub greetingRead
     my $self = shift;
 
     # Get the first line of output from the remote if possible
-    my $hGreeting = $self->{oJSON}->decode($self->{io}->lineRead(undef, undef, undef, true));
+    my $strGreeting = $self->io()->readLine(true);
```

Comment:
Yes - the new code waits but if there is an EOF exits immediately without error.  An better error will be thrown by $self->io()->error() rather than just getting an EOF error.